langchain4j-alloydb-pg/src/main/java/dev/langchain4j/store/embedding/alloydb/AlloyDBEmbeddingStore.java

```
@@ -37,52 +63,142 @@ public class AlloyDBEmbeddingStore implements EmbeddingStore<TextSegment> {
+            }
+            if (!allColumns.get(contentColumn).equalsIgnoreCase("text") || !allColumns.get(contentColumn).contains("char")) {
+                throw new IllegalStateException("Content column, is type " + allColumns.get(contentColumn) + ". It must be a type of character string.");
+            }
+            if (!allColumns.containsKey(embeddingColumn)) {
+                throw new IllegalStateException("Embedding column, " + embeddingColumn + ", does not exist.");
+            }
+            if (!allColumns.get(embeddingColumn).equalsIgnoreCase("USER-DEFINED")) {
+                throw new IllegalStateException("Embedding column, " + embeddingColumn + ", is not type Vector.");
+            }
+            if (!allColumns.containsKey(metadataJsonColumn)) {
+                metadataJsonColumn = null;
+            }
+
+            for (MetadataColumn metadataColumn : metadataColumns) {
+                if (!allColumns.containsKey(metadataColumn.getName())) {
+                    throw new IllegalStateException("Metadata column, " + metadataColumn.getName() + ", is not type Vector.");
+                }
+            }
+
+            if (ignoredColumns != null && !ignoredColumns.isEmpty()) {
+
+                Map<String, String> allColumnsCopy = allColumns.entrySet().stream()
+                        .collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
+                ignoredColumns.add(idColumn);
+                ignoredColumns.add(contentColumn);
+                ignoredColumns.add(embeddingColumn);
+
+                for (String ignore : ignoredColumns) {
+                    allColumnsCopy.remove(ignore);
+                }
+
+                metadataColumns.addAll((allColumnsCopy.entrySet().stream()
+                        .map(e -> new MetadataColumn(e.getKey(), e.getValue(), true)).collect(Collectors.toList())));
+            }
+
+        } catch (SQLException ex) {
```

Do not ignore errors, throw an error like throw new RuntimeException(String.format("Failed to execute '%s'", query), e);